privval/signer_client.go

```diff
@@ -41,6 +41,9 @@ func NewSignerClient(ctx context.Context, endpoint *SignerListenerEndpoint, chai
 
 // Close closes the underlying connection
 func (sc *SignerClient) Close() error {
+	if err := sc.endpoint.Stop(); err != nil {
+		return err
+	}
 	return sc.endpoint.Close()
```

Comment:
I think we probably want to close regardless?

```suggestion
	err := sc.endpoint.Stop()
	cerr := sc.endpoint.Close()
	if err != nil {
		return err
	}
	return cerr
```